gapic/src/main/com/google/gapid/models/Memory.java

```diff
@@ -725,7 +724,7 @@ private static void appendPointedNodes(StructNode root, Map<Long, StructNode> no
       if (root.getTypeCase() == TyCase.POINTER) {
         long pointedAddress = root.getValue().getPointer().getAddress();
         if (nodes.containsKey(pointedAddress)) {
-          root.getChildren().add(nodes.get(pointedAddress));
+          root.getChildren().add(removeExtraLayers(nodes.get(pointedAddress)));
```

Comment:
Ah, I was surprised that this line change will make a difference, because `removeExtraLayers` is already done for all the nodes previously and a second run shouldn't make any changes. 

So I looked at my code again and debugged it a little, finally found actually there's a small bug in `removeExtraLayers ` method.

I think an actual fix would be changing this line https://github.com/chrisforbes/agi/blob/7ceaa193ad3cbd339ca45a788a0923dac4cedd2d/gapic/src/main/com/google/gapid/models/Memory.java#L686 to `root = removeExtraLayers(root.children.get(0));`. Sorry I forgot to call the recursive method itself after updating root in this method, and so comes the bug.

If the bug fix is made in method `removeExtraLayers` as above, this line change will no longer be needed, you would get the same result :) And I think this would be a better and cleaner way to solve it.

Reply:
Yes, I like that better. Thanks!